src/trading/getQuote.ts

```diff
@@ -104,20 +109,63 @@ export async function getQuote(
 
   log('Getting quote...')
 
-  const quoteResponse = await orderBookApi.getQuote(quoteRequest)
+  const quote = await orderBookApi.getQuote(quoteRequest)
+
+  return {
+    quote,
+    appDataInfo,
+    orderBookApi,
+    tradeParameters,
+    slippageBpsToUse,
+  }
+}
+
+export async function getQuote(
+  _tradeParameters: TradeParameters,
+  trader: QuoterParameters,
+  advancedSettings?: SwapAdvancedSettings,
+  _orderBookApi?: OrderBookApi
+): Promise<{ result: QuoteResults; orderBookApi: OrderBookApi }> {
+  const { quote, orderBookApi, tradeParameters, slippageBpsToUse, appDataInfo } = await getQuoteRaw(
+    _tradeParameters,
+    trader,
+    advancedSettings,
+    _orderBookApi
+  )
+  const { partnerFee, slippageBps, sellTokenDecimals, buyTokenDecimals } = tradeParameters
+  const { chainId, account: from } = trader
+
+  // If AUTO slippage is used, we need to calculate the suggested slippage based on the quote)
+  if (slippageBps === undefined) {
```

Comment:
I would expect it to be controlled more explicitly, for example with a new flag 'useSmartSlippage: true' in `TradeOptionalParameters`

Reply:
But why? We want people to suggest good slippages (so their trade goes through). If not provided, currently replaces it with the default. The PR will just enhance this to have a better default.

Reply:
I would also not introduce a concept such us `smart`